vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextFieldValidationSupport.java

```diff
@@ -63,18 +66,37 @@ void setPattern(String pattern) {
      * @return <code>true</code> if the value is invalid.
      */
     boolean isInvalid(String value) {
-        final boolean isRequiredButEmpty = required
-                && Objects.equals(field.getEmptyValue(), value);
+        return checkValidity(value).isError();
+    }
+
+    ValidationResult checkValidity(String value) {
+        var requiredValidation = ValidationUtils.checkRequired(required, value,
+                field.getEmptyValue());
+        if (requiredValidation.isError()) {
+            return requiredValidation;
+        }
+
         final boolean isMaxLengthExceeded = value != null && maxLength != null
                 && value.length() > maxLength;
+        if (isMaxLengthExceeded) {
+            return ValidationResult.error(ValidationError.MAX_LENGTH_EXCEEDED);
```

Comment:
**thought:** Considering that these validation errors are defined in a shared module, I guess it might make sense to maintain their corresponding validation checks in the shared `ValidationUtils` module as well, for the sake of consistency. What do you think?

Reply:
I thought about doing that on first place, but later thought that those checks are not shared with any other component at the moment, so I decided to keep them in place. But maybe, it's a better idea to move them to `ValidationUtils` as well.